### PR TITLE
[common/meta]: refactor: use derive_more to simplify type conversion

### PR DESCRIPTION
**Pull request by drmingdrmer**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

##### [common/meta]: refactor: use derive_more to simplify type conversion
- `derive_more::From` auto impl conversion to an enum.

- feature: add `Change` to describes a state change

## Changelog

- Improvement

## Related Issues

- #2030

### CODE REVIEW

**Review by dantengsky — APPROVED** [review]
LGTM

**Comment by BohuTANG** [comment]
The CI failed due to the grcov crate, not related to this PR.

File: common/meta/types/src/change.rs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+use crate::SeqV;
+
+pub enum AddResult<T> {
+    Ok(SeqV<T>),
+    Exists(SeqV<T>),
+}
+
+/// `Change` describes a state change, including the states before and after a change.
+#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, derive_more::From)]
+pub struct Change<T>
```
Comment (ariesdevil) [inline common/meta/types/src/change.rs:28]: I think the name `Exchange` is more appropriate here.

Comment (drmingdrmer) [reply common/meta/types/src/change.rs:28]: Hm... not quite sure...

**Pull request merged**